### PR TITLE
[DEP] remove Differencer

### PR DESCRIPTION
**Pull request by TonyBagnall**
remove Differencer transformer

Note I have also removed a pipeline test that was based around pipelines with Differencer. Removing differencer made the test fail, and given the proposed restructuring, it just seems too much effort to figure out what these are doing. Happy for someone else to work on this test.

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -289,71 +289,25 @@ each time series. The `predict` method can be used to predict the cluster labels
...
 The following example shows how to use the
-[Differencer](transformations.difference.Differencer) class to extract the first
-order difference of a time series. Usage is the same for both single series and
-collection input types.
+[AutoCorrelationSeriesTransformer](transformations.series.AutoCorrelationSeriesTransformer)
+class to extract the autocorrelation terms of a time series.
 
 ```{code-block} python
->>> from aeon.transformations.difference import Differencer
+>>> from aeon.transformations.series import AutoCorrelationSeriesTransformer
```
Comment (MatthewMiddlehurst) [inline docs/getting_started.md:303]: Assume this example runs fine and thats where you got the new output. Need to remove the italy import.

File: examples/transformations/transformations.ipynb
Comment (MatthewMiddlehurst) [inline examples/transformations/transformations.ipynb:1]: Need to remove this file from the examples webpage. may be others also.

Comment (MatthewMiddlehurst) [reply examples/transformations/transformations.ipynb:1]: also, this notebook is used on the webpage index.

Comment (TonyBagnall) [reply examples/transformations/transformations.ipynb:1]: I dont think thats for this PR

**Review by MatthewMiddlehurst — APPROVED** [review]

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**